### PR TITLE
Add eslint-plugin-jest

### PR DESCRIPTION
**Pull request by andreogle**
### Changes

1. Adds [eslint-plugin-jest](https://www.npmjs.com/package/eslint-plugin-jest) to our base ESLint config and enables the recommended rules.
2. Fix the tests that don't pass the linting checks
3. Upgrade some ESLint related packages

### CODE REVIEW

File: packages/admin/test/e2e/cli.feature.ts [hunk 1]
```diff
@@ -55,8 +55,6 @@ describe('CLI', () => {
   beforeAll(() => {
     jest.setTimeout(45_000);
 
-    expect(existsSync(`${CLI_EXECUTABLE}`)).toBe(true);
```
Comment (andreogle) [inline packages/admin/test/e2e/cli.feature.ts:58]: Seems like `expect()` outside of an `it` or `test` block don't get executed: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-standalone-expect.md#rule-details

File: packages/node/src/providers/actions.test.ts [hunk 2]
```diff
@@ -164,15 +164,11 @@ describe('initialize', () => {
...
-    expect.assertions(1);
     const config = fixtures.buildConfig({ chains: [] });
     const workerOpts = fixtures.buildWorkerOptions();
-    // eslint-disable-next-line functional/no-try-statement
-    try {
-      await providers.initialize('abcdefg', config, workerOpts);
-    } catch (e) {
-      expect(e).toEqual(new Error('One or more chains must be defined in the provided config'));
-    }
+    await expect(providers.initialize('abcdefg', config, workerOpts)).rejects.toThrow(
```
Comment (andreogle) [inline packages/node/src/providers/actions.test.ts:169]: This syntax is a significant improvement imo. I wasn't aware of it before

**Review by Siegrift — APPROVED** [review]
Nice!

File: packages/node/src/providers/actions.test.ts (hunk 2 above)
Comment (acenolaza) [reply packages/node/src/providers/actions.test.ts:169]: Yes, I like that pattern too. Also if you only care about the error message then you can only pass the error string to the `.toThrow()` function.

File: packages/admin/test/e2e/cli.feature.ts (hunk 1 above)
Comment (acenolaza) [reply packages/admin/test/e2e/cli.feature.ts:58]: Then if we want to keep this check we would need to replace it with an if statement followed by a throw, correct?

Comment (acenolaza) [reply packages/admin/test/e2e/cli.feature.ts:58]: I just saw you put it in a separate `it` block which makes even more sense 🔥

**Review by acenolaza — APPROVED** [review]
LGTM :shipit:

**Pull request merged**